src/components/InfoWindow.js

```diff
@@ -0,0 +1,70 @@
+import React, { useContext } from 'react';
+import { getDistanceFromLatLonInKm } from '../helpers/Haversine';
+import { LocationsContext } from '../context/LocationsContext';
+import './InfoWindow.css';
+
+export default function InfoWindow({ $dimensionKey }) {
+  const locationsContext = useContext(LocationsContext);
+
+  const location = locationsContext.locations.filter(
```

Comment:
This is clever and seems to work, but is there not a way to pass a single location to `InfoWindow` for rendering? I’m imagining a map in a dense, historical spot (e.g., Rome) that has possibly hundreds of pins on the screen. The cost of filtering through all of the locations for every info window might be noticeable on a mobile device.

Reply:
Good point, I took this filter out and instead passed the individual location data down through props from the Map component. Does that help mitigate some of the concern here?